participant/tests/test_api.py

```diff
@@ -1,23 +1,92 @@
 import uuid
+from random import randint
 
 from django.urls import reverse
+from djangorestframework_camel_case.util import underscoreize
 from rest_framework import status
 
-from participant.models import ActionLog
+from participant.models import ActionLog, Participant
+from participant.serializer import ParticipantSerializer
 from phishtray.test.base import PhishtrayAPIBaseTest
-from ..factories import ParticipantFactory
+from ..factories import ParticipantFactory, ParticipantActionFactory, ProfileEntryFactory
 from exercise.factories import DemographicsInfoFactory
 
 
+class ParticipantAPITests(PhishtrayAPIBaseTest):
+    
+    def setUp(self):
+        super(ParticipantAPITests, self).setUp()
+    
+    def test_participant_list_block_public(self):
+        """
+        Non admin users should not be able to retrieve participant list.
+        """
+        url = reverse('api:participant-list')
+        response = self.client.get(url)
+
+        self.assertEqual(status.HTTP_403_FORBIDDEN, response.status_code)
+        
+    def test_participant_list_allow_admin(self):
+        """
+        Admin users should be able to retrieve thread list.
+        """
+        url = reverse('api:participant-list')
+        participant_count = randint(1, 50)
+        ParticipantFactory.create_batch(participant_count)
+
+        response = self.admin_client.get(url)
+        serialized = ParticipantSerializer(Participant.objects.all(), many=True)
+
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
+        self.assertEqual(participant_count, len(response.data))
+        self.assertEqual(serialized.data, underscoreize(response.data))
```

Comment:
We could camelize(serialized.data) or underscoreize(response.data) for these checks, is there any reason to do one over the other?

Reply:
Yes, I would say use `underscorize` in tests because that's the output you expect from your serializers.
Also since python is using snake_case it makes sense to compare the `underscorized` versions.
I would like to differentiate between FE and BE clients (should we ever need to call our APIs from another BE).
It's fine for know regardless.